### PR TITLE
Template holidays

### PR DESCRIPTION
**Pull request by phoebemirman**
https://github.com/AugurProject/augur/issues/5404

### CODE REVIEW

File: packages/augur-tools/src/templates-template.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { ethers } from 'ethers';
 import moment from 'moment';
 import { BigNumber } from 'ethers/utils';
+import { failures } from 'io-ts';
```
Comment (bthaile) [inline packages/augur-tools/src/templates-template.ts]: prob dont need this import

File: packages/augur-tools/src/templates-template.ts [hunk 2]
```diff
@@ -390,6 +408,31 @@ function dateNoWeekend(
     ) {
       return false;
     }
+
+    const closing = closingDateDependencies[0];
```
Comment (bthaile) [inline packages/augur-tools/src/templates-template.ts]: redundant code in templates-template and in UI validations. pull out the redundant code into a method that both can call.

**Review by bthaile — CHANGES_REQUESTED** [review]
reduce redundant code.

File: packages/augur-tools/src/templates-template.ts [hunk 3]
```diff
@@ -391,6 +437,11 @@ function dateNoWeekend(
...
+    if (closing && tellOnHoliday(inputs, input, closing)) {
+      return true;
```
Comment (bthaile) [inline packages/augur-tools/src/templates-template.ts]: should this be false?

File: packages/augur-tools/src/templates-template.ts [hunk 4]
```diff
@@ -536,8 +589,8 @@ export const isTemplateMarket = (
     }
 
     // check DATE isn't on weekend
-    if (!dateNoWeekend(template.inputs, validation.dateDependencies)) {
-      errors.push('market question date can not be on weekend');
+    if (!dateNoWeekendHoliday(template.inputs, validation.dateDependencies, validation.closingDateDependencies)) {
+      errors.push('market question date can not be on weekend or on a holiday');
```
Comment (bthaile) [inline packages/augur-tools/src/templates-template.ts:594]: need to update error message in v2 template docs as well.

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). ✌️
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/m96h3r9**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

**Review by bthaile — APPROVED** [review]
Lgtm

**Review by bthaile — APPROVED** [review]
Lgtm

**Pull request merged**